app/src/androidTest/java/org/simple/clinic/bp/BloodPressureHistoryListItemPagingSourceTest.kt

```diff
@@ -0,0 +1,208 @@
+package org.simple.clinic.bp
+
+import androidx.paging.PagingConfig
+import androidx.paging.PagingSource
+import androidx.paging.testing.TestPager
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.rx2.rxSingle
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.simple.clinic.AppDatabase
+import org.simple.clinic.TestClinicApp
+import org.simple.clinic.bp.history.adapter.BloodPressureHistoryListItem
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import org.simple.sharedTestCode.TestData
+import org.simple.sharedTestCode.util.Rules
+import org.simple.sharedTestCode.util.TestUserClock
+import org.simple.sharedTestCode.util.TestUtcClock
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+import javax.inject.Inject
+import javax.inject.Named
+
+class BloodPressureHistoryListItemPagingSourceTest {
+
+  @Inject
+  lateinit var appDatabase: AppDatabase
+
+  @Inject
+  lateinit var utcClock: TestUtcClock
+
+  @Inject
+  lateinit var userClock: TestUserClock
+
+  @Inject
+  @Named("full_date")
+  lateinit var dateFormatter: DateTimeFormatter
+
+  @Inject
+  @Named("time_for_measurement_history")
+  lateinit var timeFormatter: DateTimeFormatter
+
+  @Inject
+  lateinit var schedulersProvider: SchedulersProvider
+
+  @get:Rule
+  val ruleChain: RuleChain = Rules.global()
+
+  @Before
+  fun setup() {
+    TestClinicApp.appComponent().inject(this)
+    utcClock.setDate(LocalDate.parse("2020-01-01"))
+  }
+
+  @After
+  fun tearDown() {
+    appDatabase.bloodPressureDao().clearData()
+  }
+
+  @Test
+  fun loading_paged_data_should_work_correctly() {
```

Comment:
We don't need that with `PagingSource`. Unlike the previous `DataSource` tests, we don't have two separate functions for `PagingSource` for loading initial data and range data. It will load a page given the params, and those params are controlled by the `Pager` in this instant. 

So, as long as loading a page works. Subsequent page loads work in the similar fashion. In this test, I have tested that the page load work by loading 0-5 data instead of 0-6.